public/app/core/store.ts

```diff
@@ -35,14 +38,14 @@ export class Store {
     try {
       json = JSON.stringify(value);
     } catch (error) {
-      console.error(`Could not stringify object: ${key}. [${error}]`);
+      appEvents.emit(AppEvents.alertError, [`Could not stringify object: ${key}. [${error}]`]);
       return false;
     }
     try {
       this.set(key, json);
     } catch (error) {
       // Likely hitting storage quota
-      console.error(`Could not save item in localStorage: ${key}. [${error}]`);
+      appEvents.emit(AppEvents.alertError, [`Could not save item in localStorage: ${key}. [${error}]`]);
       return false;
     }
     return true;
```

Comment:
Add error handling. Has to be done here, as store is not throwing errors and therefore if we add try-catch block to `addQueryToRichHistory` it is not going to work.

Reply:
I would still like to lift this one level up. So I would rather have this function throw the error, handle it in the both the RichHistory helpers and in LocalStorageValueProvider where this is used. That way LocalStorageValueProvider can still keep the console.log as we don't know if the emit is appropriate there, and add the emit to rich history where we know we want it. We can do it in separate PR.